src/js/components/Select/Select.js

```diff
@@ -143,7 +141,7 @@ const Select = forwardRef(
         if (onChange) {
           event.persist();
           const adjustedEvent = event;
-          adjustedEvent.target = inputRef.current;
+          adjustedEvent.target.value = nextValue;
```

Comment:
I suspect this will be a breaking change.

Reply:
Yeah it would be a breaking change, here is what was said about it in the PR description:

> might be an issue if some people was using onChange's event.target.value as a way to obtain the previous value (see #3173 for more details). However, that is not the expected behavior and it wasn't documented as such.

Reply:
@ShimiSun I think normally `event.target.value` is used to retrieve to the current value of the input element that changed, but for Grommet's Select component it's actually the previous value.

Reply:
I think this approach is too narrowly focused, as the `event.target` will not be referring to the `<select />` DOM element but instead the `<button />` inside the drop. I think we should be using the technique we use in MaskedInput to first update the `<select />` value and then dispatching an event from it. See: https://github.com/grommet/grommet/blob/186b94413318dd7c4e30e9f8f47c124693acfd63/src/js/components/MaskedInput/MaskedInput.js#L238

Reply:
@jimenezff Any updates on this PR? It will be nice to get it on our next release.

Reply:
@ShimiSun sorry for the delay. It looks like this may have already been solved with #5155. I'll check and revert my changes to `Select.js`

Reply:
@ShimiSun I tried it but there are still issues with #3173. How about I remove the code related to that issue and we can just make this PR about type definition and the new TS story. If you agree, I can make that change today.